crates/sui-indexer-alt/src/pipeline/concurrent/commit_watermark.rs

```diff
@@ -201,9 +202,15 @@ pub(super) fn commit_watermark<H: Handler + 'static>(
                             }
 
                             Ok(updated) => {
-                                let elapsed = guard.stop_and_record();
-
                                 if updated {
+                                    let elapsed = guard.stop_and_record();
```

Comment:
I think this should still be done at the immediately after the patten match (for symmetry with the error case).